src/main/java/com/google/jenkins/plugins/k8sengine/KubectlWrapper.java

```diff
@@ -109,6 +118,9 @@ public String runKubectlCommand(String command, ImmutableList<String> args)
               .add("--kubeconfig")
               .add(kubeConfigFile.getRemote())
               .add(command);
+      if (!namespace.equals("")) {
```

Comment:
Would rather use "default" than not specifying the namespace if not present. Not specifying it is what led to the non-deterministic behavior we're trying to avoid.

Reply:
Also, this can just be set as the default value in the builder, which will allow this code to be much simpler.

Reply:
Nvm, just saw that you're including "default" in the KubeConfig's namespace field. Disregard.